pkg/sac/scope_key.go

```diff
@@ -85,10 +85,16 @@ func (k ResourceScopeKey) String() string {
 }
 
 // ResourceScopeKeys wraps the given resources in a scope key slice.
+// Note: The returned scope keys _may_ be greater than the given resources,
+// since replacing resources will be taken into account and conditionally added
+// to the returned scope keys.
 func ResourceScopeKeys(resources ...permissions.ResourceHandle) []ScopeKey {
 	keys := make([]ScopeKey, len(resources))
```

Comment:
This creates a slice of length `n`, what you need is capacity `n` because you only append. So `keys := make([]ScopeKey, 0, len(resources))`